docs/external.rst

```diff
@@ -1,68 +1,140 @@
 .. _external:
 
-Interacting with external arrays
-================================
+Exporting and importing data
+============================
 
-**External arrays** refer to ``numpy.ndarray`` or ``torch.Tensor``.
+Introduction
+------------
 
-Conversion between Taichi fields and external arrays
-----------------------------------------------------
+Taichi fields are aimed to be used in Taichi-scope.
+However, sometimes we may need to access Taichi field elements in Python-scope,
+which is not very efficient when a lot of elements within the field are required.
 
-Use ``to_numpy``/``from_numpy``/``to_torch``/``from_torch``:
+So we provide a way to export and import data in Taichi fields via NumPy arrays.
+This allows you to interact with the data in Python-scope with other packages.
+
+**Export data in Taichi fields to a NumPy array** via ``to_numpy()``.
+This allows us to export the computation result and pass it to other
+Python packages that support numpy, e.g. matplotlib.
 
 .. code-block:: python
 
-  import taichi as ti
-  import numpy as np
+   @ti.kernel
+   def my_kernel():
+      for i in x:
+         x[i] = i * 2
 
-  ti.init()
+   x = ti.field(ti.f32, 4)
+   my_kernel()
+   x_np = x.to_numpy()
+   print(x_np)  # np.array([0, 2, 4, 6])
 
-  n = 4
-  m = 7
 
-  # Taichi fields
-  val = ti.field(ti.i32, shape=(n, m))
-  vec = ti.Vector.field(3, dtype=ti.i32, shape=(n, m))
-  mat = ti.Matrix.field(3, 4, dtype=ti.i32, shape=(n, m))
+**Import data from NumPy array to Taichi fields** via ``from_numpy()``.
+This allow people to initialize Taichi fields via numpy array, e.g.:
+
+.. code-block:: python
+
+   x = ti.field(ti.f32, 4)
+   x_np = np.array([1, 7, 3, 5])
+   x.from_numpy(x_np)
+   print(x[0])  # 1
+   print(x[1])  # 7
+   print(x[2])  # 3
+   print(x[3])  # 5
+
+
+API reference
+-------------
+
+We provide interface to share data with both NumPy and PyTorch.
+
+Interacting with NumPy
+**********************
+
+.. function:: field.to_numpy()
+
+   :parameter field: (ti.field, ti.Vector.field or ti.Matrix.field) The field
+
+   :return: (np.array) The numpy array containing the currrent data in ``x``.
+
+.. function:: field.from_numpy(array)
+
+   :parameter field: (ti.field, ti.Vector.field or ti.Matrix.field) The field
+
+   :parameter array: (np.array) The numpy array containing data to initialize the field
 
-  # Scalar
-  arr = np.ones(shape=(n, m), dtype=np.int32)
 
-  val.from_numpy(arr)
+Interacting with PyTorch
+************************
 
-  arr = val.to_numpy()
+.. function:: field.to_torch(device = None)
```

Comment:
```suggestion
.. function:: field.to_torch(device=None)
```

Reply:
See `vector.rst`, it seems that `device = None` is a common style in doc..

Reply:
OK.